### ISSUE TITLE
The script encountered an issue while finding the module specification for 'main' and a socket connection problem

### ISSUE THREAD

**Issue by Blacksingh** [34j/so-vits-svc-fork#347]
It seems that the script encountered an issue while finding the module specification for 'main' and a socket connection problem. 

Happens when I try to train

(so-vitts-fork) PS C:\Users\jerme\anaconda3\envs\so-vitts-fork> svc train
[18:56:14] INFO     [18:56:14] Version: 3.8.0                                                             __main__.py:22
[18:56:18] INFO     [18:56:18] Using strategy: ddp_find_unused_parameters_true                               train.py:82
INFO: GPU available: True (cuda), used: True
           INFO     [18:56:18] GPU available: True (cuda), used: True                                    rank_zero.py:48
INFO: TPU available: False, using: 0 TPU cores
           INFO     [18:56:18] TPU available: False, using: 0 TPU cores                                  rank_zero.py:48
INFO: IPU available: False, using: 0 IPUs
           INFO     [18:56:18] IPU available: False, using: 0 IPUs                                       rank_zero.py:48
INFO: HPU available: False, using: 0 HPUs
           INFO     [18:56:18] HPU available: False, using: 0 HPUs                                       rank_zero.py:48
           WARNING  [18:56:18]                                                                           warnings.py:109
                    C:\Users\jerme\anaconda3\envs\so-vitts-fork\lib\site-packages\so_vits_svc_fork\modul
                    es\synthesizers.py:81: UserWarning: Unused arguments: {'n_layers_q': 3,
                    'use_spectral_norm': False}
                      warnings.warn(f"Unused arguments: {kwargs}")

           INFO     [18:56:18] Decoder type: hifi-gan                                                synthesizers.py:100
[18:56:19] WARNING  [18:56:19]                                                                           warnings.py:109
                    C:\Users\jerme\anaconda3\envs\so-vitts-fork\lib\site-packages\so_vits_svc_fork\utils
                    .py:187: UserWarning: Keys not found in checkpoint state dict:['emb_g.weight']
                      warnings.warn(f"Keys not found in checkpoint state dict:" f"{not_in_from}")

           INFO     [18:56:19] Loaded checkpoint 'logs\44k\G_0.pth' (iteration 0)                           utils.py:244
           INFO     [18:56:19] Loaded checkpoint 'logs\44k\D_0.pth' (iteration 0)                           utils.py:244
INFO: Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/2
           INFO     [18:56:19] Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/2                  distributed.py:244
[W ..\torch\csrc\distributed\c10d\socket.cpp:601] [c10d] The client socket has failed to connect to [kubernetes.docker.internal]:57680 (system error: 10049 - The requested address is not valid in its context.).
C:\Users\jerme\anaconda3\envs\so-vitts-fork\python.exe: Error while finding module specification for '__main__' (ValueError: __main__.__spec__ is None)

Tensor board website pops up and says "No dashboards are active for the current data set."

**Comment by Blacksingh** [comment]
Previously I was trying to install on anaconda in windows 11. My GPU is rtx3090 and needs to run in Linux Ubuntu because of

**Comment by 34j** [comment]
@allcontributors add Blacksingh bug

**Comment by Byron-Zoomry** [comment]
@Blacksingh I'm having this same issue and am unable to train. Were you able to resolve this somehow?

**Comment by Blacksingh** [comment]
Not really tbh
Try fresh with Ubuntu and install 
Nvidia Driver
Nvidia CUDA
Nvidia cudnn
Python
Tensorflow
Pytorch
Then go from there

**Comment by Byron-Zoomry** [comment]
I'm on windows. I'm already pretty in over my head trying to figure out all this github stuff to also try to install linux. Has Ubuntu given you better results?

**Comment by Blacksingh** [comment]
Yea I was in the same boat, in over my head lol But if you have an Nvidia card you’re gonna need to go the Ubuntu route, because I don’t think there’s windows support for some of the stuff I listed there for the cards to run proper for training models.When I did that, it was the closest I got, then I tried to delete n reinstall Ubuntu n everything n then I couldn’t even download PyTorch again properly so idk I had to take a break cuz of the stress it was giving me n being unable to solve Prolly gonna run off an instance my next go around Rent GPUsvast.aiSent from my iPhoneOn May 1, 2023, at 7:43 PM, Byron ***@***.***> wrote:﻿
I'm on windows. I'm already pretty in over my head trying to figure out all this github stuff to also try to install linux. Has Ubuntu given you better results?
